src/ghostfat.c

```diff
@@ -270,11 +269,11 @@ void uf2_read_block (uint32_t block_no, uint8_t *data)
     for ( uint32_t i = 0; i < FAT_ENTRIES_PER_SECTOR; ++i )
     {
       // Generate the FAT chain for the firmware "file"
-      uint32_t v = (sectionIdx * FAT_ENTRIES_PER_SECTOR) + i;
+      uint32_t v = (sectionIdx * FAT_ENTRIES_PER_SECTOR * BPB_SECTORS_PER_CLUSTER) + (i * BPB_SECTORS_PER_CLUSTER);
```

Comment:
Math at 274 looks right. Sector `v` is aligned to the cluster boundary (so it's rounded down, essentially), which means that should always be less or equal to the actual last sector.

But yes, as you specified before, it'll be simpler working with the clusters rather than sectors there.